helm lamp stack chart

fnerdman opened the pull request:
See lamp/README.md for details of the chart. Let me know if you have any suggestions to complete the pull request.

Commit:
lamp stack released for pull request

Commit:
bugfixing hostPath and persitent subpath features

viglesiasce commented:
@k8s-bot ok to test

viglesiasce commented:
/test

viglesiasce commented:
Can you add a default PW for the mysql like in the MySQL chart? This will allow the chart to install with the default values and make CI go through.

Example:
https://github.com/ROKKYT1/charts/blob/445ccf8418f19e258c54514eec5be773b1219401/stable/mysql/templates/secrets.yaml#L12

viglesiasce commented:
BTW awesome chart!!!!!!

viglesiasce reviewed (changes requested):
Make a default PW (hopefully random) for the MySQL root PW.

Commit:
disabled mysql by default

fnerdman commented:
@viglesiasce 
Random password is not a good idea. I've run into a lot of problems because updating the release would reset the password and then things like cloning the release would not work anymore. I've disabled the mysql container by default and now the build finishes successfully.

fnerdman commented:
/retest

Commit:
mysql root password enables mysql container and lead4good/httpd-fpm dependency removed by adding the httpd config files to the chart

fnerdman commented:
@viglesiasce I'd like to contribute a 'nextcloud' chart which uses the lamp chart as a dependency. How long will it take to accept this pull request?

Commit:
disabled deleting uploads directory in wp develop environment by default

Commit:
fix configmap bug

Commit:
mysql password bug

unguiculus commented on stable/lamp/Chart.yaml:
```diff
@@ -0,0 +1,7 @@
+apiVersion: v1
+description: Modular and transparent LAMP stack chart supporting PHP-FPM, Release Cloning, LoadBalancer, Ingress, SSL and lots more!
+name: lamp
+version: 0.1.0
```
Add `appVersion`.

viglesiasce replied:
We can leave this off since there is a conglomeration of app versions.

unguiculus commented on stable/lamp/README.md:
```diff
@@ -0,0 +1,314 @@
+
+## Prerequisites
+
+- Kubernetes 1.4+ with Beta APIs enabled
```
1.7+

unguiculus commented on stable/lamp/templates/deployment.yaml:
```diff
@@ -0,0 +1,902 @@
+    metadata:
+      labels:
+        app: {{ template "fullname" . }}
+      annotations:
```
Use `initContainers` field on the PodSpec instead of annotation.

unguiculus commented on stable/lamp/templates/ingress.yaml:
```diff
@@ -0,0 +1,126 @@
+{{- if .Values.ingress.enabled -}}
```
One file per resource, please.

unguiculus commented on stable/lamp/templates/ingress.yaml:
```diff
@@ -0,0 +1,126 @@
+kind: Ingress
+metadata:
+  name: {{ template "fullname" . }}-www
+  annotations:
```
Instead of wrapping various annotations in if blocks, I'd suggest you add the annotations to the configuration. This would also be more flexible.

unguiculus commented on stable/lamp/templates/pvc.yaml:
```diff
@@ -0,0 +1,24 @@
+    heritage: "{{ .Release.Service }}"
+  {{- if .Values.persistence.storageClass }}
+  annotations:
+    volume.beta.kubernetes.io/storage-class: {{ .Values.persistence.storageClass | quote }}
```
No longer use annotation for storage class. We've developed a pattern for persistence (see #1869). Please follow this approach.

unguiculus commented on stable/lamp/templates/service.yaml:
```diff
@@ -0,0 +1,66 @@
+{{- if and .Values.sftp.enabled .Values.ingress.enabled }}
+
+---
+
```
One file per resource, please.

unguiculus commented on stable/lamp/values.yaml:
```diff
@@ -0,0 +1,169 @@
+  #repository: php
+  #tag: fpm
+  pullPolicy: Always
+  fpm_enabled: true
```
Use camel case instead of underscores. Apply everywhere.

unguiculus reviewed (changes requested):
Please have a look at our best practices, especially regarding standard labels. Make sure all resources get our standard labels.
https://github.com/kubernetes/helm/blob/master/docs/chart_best_practices/labels.md

Note that we've started namespacing template (see #1785). Please apply this to your chart.

Please update the PR with latest master.

unguiculus commented:
Marking this as stale. Please update within one week.

Commit:
Merge branch 'master' of https://github.com/kubernetes/charts

Commit:
moved init containers from beta annotation into spec

Commit:
split up ingress file and added custom annotations

Commit:
moved storage class annotation to stable

Commit:
split up service.yaml

Commit:
camel casing variables

Commit:
namespace defined templates

Commit:
values.yaml best practice

Commit:
labels best practice

Commit:
fixed ingress annotations

Commit:
fixed forgotten fullname helper

fnerdman commented:
@unguiculus 
I've fixed almost all of the issues. There are two things to note:

- appVersion: since there are different "apps" being used, such as php, mysql etc. and the versions can change via values, i wouldn't know which one to chose

- I added custom annotations to ingress however some of the annotations inside if clauses i left since they access template variables / have a broader spectrum and need more logic elsewhere

unguiculus commented on stable/lamp/templates/configmap-httpd.yaml:
```diff
@@ -0,0 +1,15 @@
+metadata:
+  name: {{ template "lamp.fullname" . }}-httpd
+  labels:
+    app: {{ template "lamp.fullname" . }}
```
Should be `app: {{ template "lamp.name" . }}`. Please update everywhere.

unguiculus commented on stable/lamp/templates/service-sftp.yaml:
```diff
@@ -0,0 +1,23 @@
+    protocol: TCP
+    name: sftp
+  selector:
+    app: {{ template "lamp.fullname" . }}
```
Add `release` label to selector.

unguiculus commented on stable/lamp/templates/service.yaml:
```diff
@@ -0,0 +1,46 @@
+    name: phpmyadmin
+  {{ end }}
+  selector:
+    app: {{ template "lamp.fullname" . }}
```
Add `release` label to selector.

Commit:
updated app labels and service selector release labels

fnerdman commented:
@unguiculus updated

unguiculus commented:
@viglesiasce Could you have another look, please?

viglesiasce commented on stable/lamp/values.yaml:
```diff
@@ -0,0 +1,337 @@
+  ## `ingress.kubernetes.io/proxy-body-size
+  annotations:
+    ingress.kubernetes.io/proxy-body-size: "50m"
+    kubernetes.io/ingress.class: nginx
```
Should leave these commented so that you can use the default ingress in the cluster, unless the proxy-body-size and NGINX is required.

fnerdman replied:
Indeed, nginx ingress controller is not mandatory although some features such as ssl are only working with nginx controller. I've commented that part out and added explanations regarding nginx controller requirements to the values which need the controller.

viglesiasce commented:
/ok-to-test

Commit:
commented nginx specific ingress annotations

Commit:
linter error: removed empty line

viglesiasce commented:
awesome in that case:

/lgtm

@unguiculus all set on my end, merge when you are ready
